lexicon/modification.py

```diff
@@ -0,0 +1,156 @@
+                # sort by first orth entry
+                lemma_dict[lemma.orth[0]] = lemma
+            lex.lemmas = [lemma_dict[key] for key in sorted(lemma_dict.keys())]
+        else:
+            lex.lemmas = self.static_lexicon.lemmas
+
+        write_xml(self.out_bliss_lexicon.get_path(), lex.to_xml())
+
+
+class MergeLexiconJob(Job):
+    """
+    Merge multiple bliss lexica into a single bliss lexicon.
+
+    Phonemes and lemmas can be individually sorted alphabetically or kept as is.
+
+    When merging a lexicon with a static lexicon, putting the static lexicon first
+    and only sorting the phonemes will result in the "typical" lexicon structure.
+    """
+
+    def __init__(self, bliss_lexica, sort_phonemes, sort_lemmas, compressed=False):
+        """
+        :param list[Path] bliss_lexica: list of bliss lexicon files (plain or gz)
+        :param bool sort_phonemes: sort phoneme inventory alphabetically
+        :param bool sort_lemmas: sort lemmas alphabetically based on first orth entry
+        :param bool compressed: compress final lexicon
+        """
+        self.lexica = bliss_lexica
+        self.sort_phonemes = sort_phonemes
+        self.sort_lemmas = sort_lemmas
+
+        self.out_bliss_lexicon = self.output_path(
+            "lexicon.xml.gz" if compressed else "lexicon.xml"
+        )
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        merged_lex = lexicon.Lexicon()
+
+        lexica = []
+        for lexicon_path in self.lexica:
+            lex = lexicon.Lexicon()
+            lex.load(lexicon_path.get_path())
+            lexica.append(lex)
+
+        # combine the phonemes
+        merged_phonemes = OrderedDict()
+        for lex in lexica:
+            for symbol, variation in lex.phonemes.items():
+                if symbol in merged_phonemes.keys():
+                    assert variation == merged_phonemes[symbol], (
+                        "conflicting phoneme variant for phoneme: %s" % symbol
+                    )
+                else:
+                    merged_phonemes[symbol] = variation
+
+        if self.sort_phonemes:
+            sorted_phoneme_list = [
+                (k, merged_phonemes[k]) for k in sorted(merged_phonemes.keys())
+            ]
+            for phoneme_tuple in sorted_phoneme_list:
+                merged_lex.add_phoneme(
+                    symbol=phoneme_tuple[0], variation=phoneme_tuple[1]
+                )
+        else:
+            merged_lex.phonemes = merged_phonemes
+
+        # combine the lemmas
+        if self.sort_lemmas:
+            lemma_dict = {}
+            for lex in lexica:
+                for lemma in lex.lemmas:
+                    # sort by first orth entry
+                    lemma_dict[lemma.orth[0]] = lemma
+            merged_lex.lemmas = [lemma_dict[key] for key in sorted(lemma_dict.keys())]
+        else:
+            for lex in lexica:
+                merged_lex.lemmas.extend(lex.lemmas)
```

Comment:
If a lemma existed in more than one lexicon, then it would now appear twice in the merged lexicon. This is different from the sorted case.